scrapy/downloadermiddlewares/httpproxy.py

```diff
@@ -26,7 +26,7 @@ def _get_proxy(self, url, orig_type):
         proxy_url = urlunparse((proxy_type or orig_type, hostport, '', '', '', ''))
 
         if user:
-            user_pass = '%s:%s' % (unquote(user), unquote(password))
+            user_pass = to_bytes('%s:%s' % (unquote(user), unquote(password)))
```

Comment:
I just found out about [RFC 7617](https://tools.ietf.org/html/rfc7617)

> The original definition of this authentication scheme failed to
   specify the character encoding scheme used to convert the user-pass
   into an octet sequence.  In practice, most implementations chose
   either a locale-specific encoding such as ISO-8859-1 ([ISO-8859-1]),
   or UTF-8 ([RFC3629]).  For backwards compatibility reasons, this
   specification continues to leave the default encoding undefined, as
   long as it is compatible with US-ASCII (mapping any US-ASCII
   character to a single octet matching the US-ASCII character code).

Servers can ask for UTF-8 but this happens in challenges only, using `charset` :

> In challenges, servers can use the 'charset' authentication parameter
   to indicate the character encoding scheme they expect the user agent
   to use when generating "user-pass" (a sequence of octets).  This
   information is purely advisory.
> The only allowed value is "UTF-8"; it is to be matched case-
   insensitively (see [RFC2978], Section 2.3).  It indicates that the
   server expects character data to be converted to Unicode
   Normalization Form C ("NFC"; see Section 3 of [RFC5198]) and to be
   encoded into octets using the UTF-8 character encoding scheme
   ([RFC3629]).